### DISCUSSION TITLE
Feedback screen info / Feedback screen

### DISCUSSION THREAD

**Discussion by aweell** [Telefonica/mistica-design#1110] [Proposals]
Actualmente en las specs de feedbacks se indica que el asset del componente InfoFeedbackScreen es libre y puede modificarse, sin embargo el componente no permite la misma configuración en desarrollo.

![Screenshot 2023-03-07 at 17 28 13](https://user-images.githubusercontent.com/44420072/223485620-2afbde60-5b9d-4d45-a2e1-adb45ce598a6.png)

Con esto me surgen las siguientes dudas:

* Deberían las librerías tener el componente FeedbackScreen con la prop para modificar el icon y eliminarlo del FeedbackInfo? 
* Debería permitirse que el FeedbackInfo tenga una prop para modificar su icon en desarrollo?

Quizás evitaría malentendidos a la hora del handoff que el componente haga match 1:1 con el nombre que tiene en implementación + posibilidades de modificación.

* [Ejemplo InfoFeedbackScreen ->](https://rebrand.ly/kk5chk0)
* [Ejemplo FeedbackScreen ->](https://rebrand.ly/r3zb8va)

[Playroom traga si le metes la prop Icon, pero typescript no miente...]

![Screenshot 2023-03-07 at 17 46 11](https://user-images.githubusercontent.com/44420072/223490706-4f746f92-1b1b-4546-86bf-47a8e45e2abe.png)

**Comment by yceballost** [comment]
Por dar un poco de contexto, que este componente tiene canas ya

El feedback screen es un componente padre **en web** para gestionar mejor las tres variantes, pero desde diseño nunca se planteo el "feedback para gobernarlos a todos" sino que desde el principio se quiso tener tres tipologías de feedback screen para cada caso.

- Success
- Error
- Informative

Desconozco si el componente en web es público, pero igual debería ser privado. (Habría que mirar si tiene usos)

En el principio, el asset era fijo en los tres casos. En algún momento se dijo de abrir la posibilidad de poder cambiar el icono del feedback info porque es cierto que ese componente tiene cabida en diferentes casos de uso que es informativo pero que el icono refuerza el mensaje. Sin embargo, en el caso de success y error nunca se permitió el cambio de icono (y personalmente nunca lo permitiría) porque son componentes con un caso de uso super claro. 

Mi pregunta, por qué querrías tener un icono custom en el success y el error?

**Reply by aweell** [reply]
> por qué querrías tener un icono custom en el success y el error?

Es cierto que añadiendo el FeedbackScreen abres la puerta a que estos otros dos componentes puedan no utilizarse y se le metan iconos custom para los casos de uso de error y success. Aquí no puedo estar más de acuerdo, no debería estar abierto a esas modificaciones.

> En algún momento se dijo de abrir la posibilidad de poder cambiar el icono del feedback info porque es cierto que ese componente tiene cabida en diferentes casos de uso que es informativo pero que el icono refuerza el mensaje.

Este es el caso global del que surge esta duda:

![Screenshot 2023-03-08 at 08 30 58](https://user-images.githubusercontent.com/44420072/223649471-162e59e1-bced-4ca2-b57c-94675deaa144.png)


He estado indagando y a la pregunta: Debería permitirse que el FeedbackInfo tenga una prop para modificar su icon en desarrollo?

La respuesta es si, archivo la discussion por repetida.

https://github.com/Telefonica/mistica-design/issues/699
https://jira.tid.es/browse/WEB-742
